llvm/lib/Support/Unix/Threading.inc

```diff
@@ -33,6 +33,13 @@
 #include <pthread_np.h> // For pthread_getthreadid_np() / pthread_set_name_np()
 #endif
 
+// Must be included after Threading.inc to provide definition for llvm::thread
+// because FreeBSD's condvar.h (included by user.h) misuses the "thread"
+// keyword.
+#ifndef __FreeBSD__
```

Comment:
I wonder if we should just undef thread instead on FreeBSD. I think that we can just do this for now, it is more conservative.